### PR TITLE
T96 coverage windows2

### PR DESCRIPTION
**Pull request by dlahtine**
Added an endpoint for fetching coverage windows.

To test: make an GET request to the /api/objects/<object_identifier>/coverage endpoint.

object identifier can be an NAIF ID, or the all-caps SPICE name for an object.

### CODE REVIEW

**Comment by zawata** [comment]
remake this to not merge into master

**Comment by dlahtine** [comment]
This is conflicting with other changes. needs refactoring.

File: FlaskServer.py [hunk 1]
```diff
@@ -41,9 +41,54 @@ def get_main_id():
...
+    windows_piecewise = []
+    for k in kernels:
+        spy.main_file = k
+        try:
+            windows_piecewise += spy.get_coverage_windows(NAIF_id)
+            windows_piecewise.sort(key=lambda x: x[0])
+        except spyce.InternalError:
+            #object does not exist in this kernel.
+            pass
+    coverage_window['start'] = windows_piecewise[0][0]
```
Comment (zawata) [inline FlaskServer.py]: you catch the internal error and pass which is fine, but then go on to operate on a potentially empty array. 
Passing 399 to the function throws an IndexError.
check `len(windows_piecewise) > 0`

**Review by zawata — CHANGES_REQUESTED** [review]
rebased on dev. logic is fine.

address the comments

**Review by zawata — APPROVED** [review]

**Comment by zawata** [comment]
I fixed it. Merging

**Pull request merged**